src/ol/PluggableMap.js

```diff
@@ -605,6 +605,9 @@ class PluggableMap extends BaseObject {
    * Clean up.
    */
   disposeInternal() {
+    this.controls.clear();
+    this.interactions.clear();
+    this.overlays_.clear();
```

Comment:
The link interaction modifies the URL when added to a map.  The interaction removes its search parameters from the URL when it is removed from a map.  The tests make use of `map.dispose()` to clean up, and they rely on the URL not having extra junk in it, so `map.dispose()` now removes interactions (and controls and overlays) so that additional cleanup can be done.